packages/valory/connections/test_libp2p/tests/test_dht.py

```diff
@@ -281,3 +305,40 @@ def teardown(self):
         """Clean up after test case run."""
         self.unset_agent_context()
         self.run_cli_command("delete", self.agent_name)
+
+
+test_classes = [
+    Libp2pConnectionDHTRelay,
+    Libp2pConnectionDHTDelegate,
+    Libp2pConnectionDHTRelayAEACli,
+    Libp2pConnectionDHTDelegateAEACli,
+]
+
+
+@dataclass
+class TestCaseConfig:
+    """TestCase"""
+
+    name: str
+    nodes: List[NodeConfig]
+    use_local_acn: bool = False
+
+
+# dynamically create tests
+for base_cls in test_classes:
+    for test_case in (
+        TestCaseConfig("Local", local_nodes, True),
+        TestCaseConfig("Public", public_nodes),
+    ):
+        name = f"Test{test_case.name}{base_cls.__name__}"
+
+        if test_case.use_local_acn:
+            bases = base_cls, ACNWithBootstrappedEntryNodes
+            test_cls = type(name, bases, {})
+        else:
+            test_cls = type(name, (base_cls,), {})
+            test_cls.pytestmark = skip_if_ci_marker
+
+        test_cls.__name__ = name
+        test_cls.nodes = test_case.nodes
+        globals()[test_cls.__name__] = test_cls
```

Comment:
we either:
- add the docker image autouse fixture to bases if `use_local_acn`
- or optionally add the `skip_if_ci_marker` to not run tests against non-local ACN